### PR TITLE
Release dev2

### LINKED ISSUE #117: Logging code type error
https://github.com/spoonsso/dannce/blob/9bcb3e38043369c23a2cc666d3a665e7de1c3ddb/dannce/engine/inference.py#L41

Line is attempting to concatenate n_frame (an integer) to prepend_long (a string)

### LINKED ISSUE #138: dannce-predict loss very small, but result same like normal but shift
hi @davidhildebrand @spoonsso @data-hound 

The loss of the prediction result is very low, and the relative position of each label is correct, but the result seems to be shifted as a whole, which does not match the video image. But the result of com-predict looks normal.
Thank you so much for your time...
`
viewGui = View3D(params, videos, skeleton);

pts3d = load('F:\dannce\demo\mouse20230129\DANNCE\predict_results\save_data_MAX0.mat')

viewGui.loadFrom3D(pts3d.pred) 
`

截屏2023-02-08 上午1 12 23

截屏2023-02-08 上午1 31 39

### PR DESCRIPTION
**Pull request by data-hound**
New release for DANNCE

## New Features
* Mirror augmentation added 
    -  Mirror the images in half of the samples of the training set
    - If mirror augmentation is used, the right_keypoints indices and left_keypoints indices must be specified as well. 
        For the skeleton, ['RHand, 'LHand', 'RFoot', 'LFoot'], 
        &emsp; set right_keypoints: [0, 2] and 
        &emsp; left_keypoints: [1, 3] in the **config file**
* Options for layer and instance normalization added.
* More Losses available: `Huber`, `logcosh`, `gaussian cross entropy` 
* `n-rand-views` configured to choose a subset of camera views for training/testing on weights trained for a lesser number of views. <br>(***Note:*** the number of views for which the weights have been trained should match the number of views specified with n-rand-views)
* `AVG+MAX` training introduced where both the mean of heatmaps and the pmax of heatmaps are used for loss computation.
* Saving prediction targets when running in MAX mode enabled

## Codebase Improvements
* Cluster scripts have now been remodelled to use yaml config files specific to each cluster instead of a set of shell scripts 
    * Sbatch submissions for dannce and com training and prediction have been implemented.
* Logging statements added throughout the code base with options to change the `log level` in the config (closes #117, closes #138)
* DANNCE now supports `mat73` format files for label3d and predictions 
* Python based scripts for plotting predictions on videos added
* Newer tests using function name parsing
* Uniform nomenclature for `results_dir` across codebase
* Callbacks moved to a new module
* Code refactored and reformatted across the codebase
* Trimming dependencies in multiple modules
* Saving pred targets moved to end of training

## Bug Fixes and Minor Enhancements
* Allow multiple read attempts when loading video
* Added sigmoid normalization for pmax in `dannce-predict`
* Introduced closing videos from prev experiments
* Remove GPU placement during inference
* Model summary printed during training to help debug mismatches
* Save training data into npy volumes and cleanup unused
* Remove hardcoded file names in `multi-gpu`

### CODE REVIEW

**Comment by spoonsso** [comment]
@data-hound can you please test out the README install instructions in a fresh conda environment and then run all tests? thanks!

**Comment by data-hound** [comment]
Done. The install instructions work and all tests under `/tests` are passing

> @data-hound can you please test out the README install instructions in a fresh conda environment and then run all tests? thanks!
